docs/src/man/querying_frameworks.md

```diff
@@ -0,0 +1,129 @@
+# Querying frameworks
+
+## Query.jl
+
+The [Query.jl](https://github.com/davidanthoff/Query.jl) package provides advanced data manipulation capabilities for `DataFrames` (and many other data structures). This section provides a short introduction to the package, the [Query.jl documentation](http://www.david-anthoff.com/Query.jl/stable/) has a more comprehensive documentation of the package.
+
+To get started, install the Query.jl package:
+
+```julia
+Pkg.add("Query")
+```
+
+A query is started with the `@from` macro and consists of a series of query commands. Query.jl provides commands that can filter, project, join, group, flatten and group data from a `DataFrame`. A query can return an iterator, or one can materialize the results of a query into a variety of data structures, including a new `DataFrame`.
+
+A simple example of a query looks like this:
+
+```@meta
+DocTestSetup = quote
+    using DataFrames, Query
+end
+```
+
+```jldoctest
+using DataFrames, Query
+
+df = DataFrame(name=["John", "Sally", "Roger"], age=[54., 34., 79.], children=[0, 2, 4])
+
+q1 = @from i in df begin
+     @where i.age > 40
+     @select {number_of_children=i.children, i.name}
+     @collect DataFrame
+end
+
+println(q1)
+
+# output
+
+2×2 DataFrames.DataFrame
+│ Row │ number_of_children │ name    │
+├─────┼────────────────────┼─────────┤
+│ 1   │ 0                  │ John  │
+│ 2   │ 4                  │ Roger │
+```
+
+```@meta
```

Comment:
OK.

@MichaelHatherly Do you think it would be possible to have a way of specifying that a series of doctests goes together and should share state? Looks like it could also be useful to allow setting a `DocTestSetup` block to be used only by the following doctests, but not the others.